roms_tools/setup/cdr_release.py

```diff
@@ -0,0 +1,481 @@
+    name : str
+        Unique identifier for the release.
+    lat : float
+        Latitude of the release location in degrees North. Must be between -90 and 90.
+    lon : float
+        Longitude of the release location in degrees East.
+    depth : float
+        Depth of the release in meters. Must be non-negative.
+    hsc : float
+        Horizontal scale (standard deviation) of the release in meters. Must be non-negative. Defaults to 0.0.
+    vsc : float
+        Vertical scale (standard deviation) of the release in meters. Must be non-negative. Defaults to 0.0.
+    times : list of datetime.datetime, optional
+        Explicit time points for volume fluxes and tracer concentrations. Defaults to [self.start_time, self.end_time] if None.
+
+        Example: `times=[datetime(2022, 1, 1), datetime(2022, 1, 2), datetime(2022, 1, 3)]`
+
+    volume_fluxes : float or list of float, optional
+
+        Volume flux(es) of the release in m³/s over time.
+
+        - Constant: applies uniformly across the entire simulation period.
+        - Time-varying: must match the length of `times`.
+
+        Example:
+
+        - Constant: `volume_fluxes=1000.0` (uniform across the entire simulation period).
+        - Time-varying: `volume_fluxes=[1000.0, 1500.0, 2000.0]` (corresponds to each `times` entry).
+
+    tracer_concentrations : dict, optional
+
+        Dictionary of tracer names and their concentration values. The concentration values can be either
+        a float (constant in time) or a list of float (time-varying).
+
+        - Constant: applies uniformly across the entire simulation period.
+        - Time-varying: must match the length of `times`.
+
+        Default is an empty dictionary (`{}`) if not provided.
+
+        Example:
+
+        - Constant: `{"ALK": 2000.0, "DIC": 1900.0}`
+        - Time-varying: `{"ALK": [2000.0, 2050.0, 2013.3], "DIC": [1900.0, 1920.0, 1910.2]}`
+        - Mixed: `{"ALK": 2000.0, "DIC": [1900.0, 1920.0, 1910.2]}`
+
+    fill_values : str, optional
+
+        Strategy for filling missing tracer concentration values. Options:
+
+        - "auto" (default): automatically set values to non-zero defaults
+        - "zero": fill missing values with 0.0
+    """
+
+    times: Optional[List[datetime]] = None
+    volume_fluxes: Union[NonNegativeFloat, List[NonNegativeFloat], Flux] = Field(
+        default=0.0
+    )
+    tracer_concentrations: Optional[
+        Dict[str, Union[NonNegativeFloat, List[NonNegativeFloat]]]
+    ] = None
+    fill_values: Literal["auto", "zero"] = "auto"
+
+    @model_validator(mode="after")
+    def _postprocess(self) -> "VolumeRelease":
+        if self.times is None:
+            self.times = []
+        num_times = len(self.times)
+
+        if self.tracer_concentrations is None:
+            self.tracer_concentrations = {}
+
+        defaults = get_tracer_defaults()
+        for tracer_name in defaults.keys():
+            if tracer_name not in self.tracer_concentrations:
```

Comment:
consider inverting to reduce nesting...

```py
if tracer_name in self.tracer_concentrations:
  continue

```

or avoid altogether by prefiltering:

```py
filtered_keys = set(defaults.keys()).difference(self.tracer_concentrations)
for tracer_name in filtered_keys:
  ...
```

Reply:
done in d2aeeaf